dask/dataframe/backends.py

```diff
@@ -185,7 +185,7 @@ def meta_nonempty_dataframe(x):
     idx = meta_nonempty(x.index)
     dt_s_dict = dict()
     data = dict()
-    for i, c in enumerate(x.columns):
+    for i, _ in enumerate(x.columns):
```

Comment:
Since we just want the position of each column, I think something like (note: I've not tested the suggestion below:

```suggestion
    for i in range(len(x.columns)):
```

would be more clear for readers